view/src/components/deal.js

```diff
@@ -0,0 +1,479 @@
+          url: `/todo/${this.state.todoId}`, // NOT CONNECTED TO CRUD YET
+          method: "put",
+          data: userTodo,
+        };
+      } else {
+        options = {
+          url: "/todo",
+          method: "post",
+          data: userTodo,
+        };
+      }
+      axios.defaults.headers.common = { Authorization: `${this.getAuth()}` };
+      axios(options)
+        .then(() => {
+          // Page state
+          this.setState({ open: false });
+          window.location.reload();
+        })
+        .catch((error) => {
+          // Page state
+          this.setState({ open: true, errors: error.response.data });
+          console.error(error);
+        });
+    };
+
+    const handleViewClose = () => {
+      // Page state
+      this.setState({ viewOpen: false });
+    };
+
+    const handleDialogClose = (event) => {
+      // Page state
+      this.setState({ open: false });
+    };
+
+    if (this.state.uiLoading === true) {
+      return (
+        <main className={classes.content}>
+          {this.state.uiLoading && <CardSkeletons classes={classes} />}
+        </main>
+      );
+    } else {
+      return (
+        <main className={classes.content}>
+          <div className={classes.toolbar} />
+
+          <Fab
+            color="primary"
+            className={classes.floatingButton}
+            aria-label="Add Produce"
+            onClick={handleAddClick}
+          >
+            <AddIcon />
+          </Fab>
+          <Dialog
+            fullScreen
+            open={open}
+            onClose={handleDialogClose}
+            TransitionComponent={Transition}
+          >
+            <AppBar className={classes.appBar}>
+              <Toolbar>
+                <IconButton
+                  edge="start"
+                  color="inherit"
+                  onClick={handleDialogClose}
+                  aria-label="close"
+                >
+                  <CloseIcon />
+                </IconButton>
+                <Typography variant="h6" className={classes.title}>
+                  {this.state.buttonType === "Edit"
+                    ? "Edit Deal"
+                    : "Create a new Deal"}
+                </Typography>
+                <Button
+                  autoFocus
+                  color="inherit"
+                  onClick={handleSubmit}
+                  className={classes.submitButton}
+                >
+                  {this.state.buttonType === "Edit" ? "Save" : "Submit"}
+                </Button>
+              </Toolbar>
+            </AppBar>
+            <Container maxWidth="lg">
+              <DealStepper alert={this.props.alert} />
+            </Container>
+          </Dialog>
+          <Container maxWidth="lg">
+            <Grid container spacing={2} alignItems="center">
+              <Grid item xs={12}>
+                <div className={classes.search}>
+                  <div className={classes.searchIcon}>
+                    <SearchIcon />
```

Comment:
Does the div that contains this need an aria-label? If it's a purely decorative icon, it should probably be `aria-hidden="true"`